src/Telegram/Types/Chat/ChatMember.php

```diff
@@ -20,31 +17,11 @@
  *
  * @see https://core.telegram.org/bots/api#chatmember
  */
-class ChatMember extends BaseType
+#[ChatMemberResolver]
+abstract class ChatMember extends BaseType
 {
-    use ChatMemberOwner;
-    use ChatMemberAdministrator;
-    use ChatMemberMember;
-    use ChatMemberRestricted;
-    use ChatMemberLeft;
-    use ChatMemberBanned;
-
     /**
      * Information about the user
      */
     public User $user;
-
-
-    public function getType(): ?string
-    {
-        return match ($this->status) {
-            ChatMemberStatus::CREATOR => ChatMemberType::OWNER,
-            ChatMemberStatus::ADMINISTRATOR => ChatMemberType::ADMINISTRATOR,
-            ChatMemberStatus::MEMBER => ChatMemberType::MEMBER,
-            ChatMemberStatus::RESTRICTED => ChatMemberType::RESTRICTED,
-            ChatMemberStatus::LEFT => ChatMemberType::LEFT,
-            ChatMemberStatus::KICKED => ChatMemberType::BANNED,
-            default => throw new RuntimeException('Invalid ChatMemberStatus type'),
-        };
-    }
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** removing this will introduce a breaking change, please reimplement this according to the new logic‏